build.gradle.kts

```diff
@@ -258,13 +258,63 @@ configure(subprojectsWithoutToolAndSmokeTestProjects) {
     apply(plugin = "ch.tutteli.gradle.plugins.dokka")
     apply(plugin = "ch.tutteli.gradle.plugins.publish")
 
-    project.extensions.getByType<ch.tutteli.gradle.plugins.publish.PublishPluginExtension>().apply {
+    extensions.getByType<ch.tutteli.gradle.plugins.publish.PublishPluginExtension>().apply {
         resetLicenses("EUPL-1.2")
     }
+
+    val languageSuffixes = setOf("-en_GB", "-de_CH")
+    if (languageSuffixes.any { name.contains(it) }) {
+        extensions.getByType<PublishingExtension>().apply {
+            val pubs = publications.toList()
+            publications {
+                pubs
+                    .filterIsInstance<MavenPublication>()
```

Comment:
pubs.withType<MavenPublication>()

Reply:
using `publications.toList()` and filterIsIsntance on purpose because I don't want that the newly created publication is re-processed (I think we would even run into a ConcurrentModificationException or the like)

Reply:
>I don't want that the newly created publication is re-processed

It creates a fragile logic that forces the reader to determine the created publications' order.

It would be easier to reason about if the relocation was like "ok, whenever a new publication arrives, we add a corresponding relocation".
The scripts that expect a specific initialization order are fragile, and they are hard to maintain.

Reply:
if we use `withType` then behind the scene gradle does something along the line of
```
publications.forEach {
  ...
}
```
if we do a create within this forEach, then we run into a ConcurrentModificationException.
See for instance (random google hit: https://rollbar.com/blog/java-concurrentmodificationexception/)

Reply:
The following works for me:

```kotlin
        configure<PublishingExtension> {
            publications.withType<MavenPublication>()
                .matching { !it.name.endsWith("-relocation") }
                .all {
                    val pub = this
                    val artifactIdWithoutLanguage = languageSuffixes.fold(artifactId) { id, suffix ->
                        id.replace(suffix, "")
                    }
                    artifactId = artifactIdWithoutLanguage
                    if (!pub.artifactId.endsWith("metadata")) {
                        val oldArtifactId = if (pub.artifactId == name) {
                            "$name-common"
                        } else if (pub.artifactId.endsWith("-jvm")) {
                            name
                        } else {
                            pub.artifactId
                        }

                        pub.artifactId = artifactIdWithoutLanguage

                        publications.create<MavenPublication>("${pub.name}-relocation") {
                            pom {
                                // Old artifact coordinates
                                groupId = pub.groupId
                                artifactId = oldArtifactId
                                version = pub.version

                                distributionManagement {
                                    relocation {
                                        // New artifact coordinates
                                        groupId.set(pub.groupId)
                                        artifactId.set(artifactIdWithoutLanguage)
                                        version.set(pub.version)
                                        message.set("artifactId has changed, removed -en_GB as we drop support for a translated report and jvm has now the suffix -jvm and common no longer has the suffix -common")
                                    }
                                }
                            }
                        }
                    }
                }
```

Reply:
An alternative option would be

```kotlin
// We want to create relocations for all the publications, so we wait for them to settle
afterEvaluate {
  val pubs = publications.withType<MavenPublication>().toList().
  // ...
}
```

However, it would be better to avoid `afterEvaluate` when possible.

Reply:
good to know, I also prefer the one without afterEvaluate. I first used withType but used the overload with the lamba and used an if inside instead of matching (is nicer). I wonder why I ran into the concurrent MidificationException. 🤔
anyway, will adopt the code and merge afterwards